modules/magnolia/src/main/scala/pureconfig/module/magnolia/MagnoliaConfigWriter.scala

```diff
@@ -0,0 +1,52 @@
+package pureconfig.module.magnolia
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+import _root_.magnolia._
+import com.typesafe.config.{ ConfigValue, ConfigValueFactory }
+import pureconfig._
+import pureconfig.error.ConfigReaderException
+import pureconfig.generic.{ CoproductHint, ProductHint }
+
+/**
+ * An object containing Magnolia `combine` and `dispatch` methods to generate `ConfigWriter` instances.
+ */
+object MagnoliaConfigWriter {
+
+  def combine[A](ctx: CaseClass[ConfigWriter, A])(implicit hint: ProductHint[A]): ConfigWriter[A] =
+    if (ctx.typeName.full.startsWith("scala.Tuple")) combineTuple(ctx)
+    else if (ctx.isValueClass) combineValueClass(ctx)
+    else combineCaseClass(ctx)
+
+  def combineCaseClass[A](ctx: CaseClass[ConfigWriter, A])(implicit hint: ProductHint[A]): ConfigWriter[A] = new ConfigWriter[A] {
```

Comment:
As with the `MagnoliaConfigReader` object, I think we can make all these `combine*` methods private and only expose `combine` and `dispatch`.